### ISSUE TITLE
[4.1] wifi password lost after each update

### ISSUE THREAD

**Issue by cpcbegin** [recalbox/recalbox-os#1221]
Every time I update to a new version, I can't access to internet because my wifi password changes to another strange string.
The first time I try to change add some ramdom caracters and I can't connect.
After several times deleting the password, I can change the password and connect to internet.

**Comment by RupertAvery** [comment]
Yeah, I noticed this as well. Is it that the encrypted password is not being decrypted upon being read?

Can the encrypted password be stored in a separate variable? Although unlikely nothing can stop somebody from having their password start with enc:

4.0.0 firmwares can use the password variable, 4.1.x can use password if it exists or a new passwordenc variable.

**Comment by garnieretienne** [comment]
I've the same issue but cannot connect using the WIFI anymore. I've noticied I've lost the network connection after the update, and even if I re-enter my wifi password, the wifi never connect.

I have a recalbox v3 with internal wifi.

**Comment by substring** [comment]
Issue identified, fixed on developpement builds

**Comment by rockaddicted** [comment]
Is this problem fixed on last update ?

**Comment by pablomacaluso** [comment]
@rockaddicted its not :( im still having this issue.

**Comment by cpcbegin** [comment]
I haven't got this problem again in latest updates.

**Comment by substring** [comment]
it has been solved with the update of 23rd of march i believe

**Comment by sismo99** [comment]
I got 2 RB pi 3 i configured the wifi in one and works great (Lets say RBPi Number 1) but when i put the sd card  in the other one (RBPi Number 2) i got the ENC problem with the password and when i return the SD Card to the original RBPi 3 (Number 1) the Wifi works well again

I tried installing and configring another fresh image  but this Time on the RBPi Number 2 and works but when i move the SD Card to the other One )Number 1 Now i got the encryption Problem Again 

So I noticed that the problem is when i change The MicroSD I configured in One RBPi to Another one the problem is that there is no way to fix the wifi config to work in the receiver RBpi just fresh instllation and configuring the awifi rom there

Can you say if there is a fix for that

thanks

**Comment by rockaddicted** [comment]
WiFi Key's encryption algorithm is using specifics values on each rpi.
So a key encrypted on rpi number 1 can't be read on rpi number 2 etc... Even if the 2 rpi are of the same generation.
1 SD card = 1 rpi once wifi configured.

**Comment by sismo99** [comment]
ok i undesrtand that and thanks for your answer Now the next problem is that when i put the Card in RBPi Numer 2 and delete the password and put it again (You know to use it in this new RBPi)  it doesn´t work either i put the password and don´t connect i have to delete the password several times and put it again and restart the RBPi to get the WiFi to work (Not an exact number of times)

I will make some tests To try to get the exact number of times 

I hope you can fix that thanks
